components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/dao/impl/PolicyDAOImpl.java

```diff
@@ -90,71 +111,96 @@ public Policy getPolicy(String policyLevel, String policyName) throws APIMgtDAOE
-                                    policy.getDescription(), policy.getDefaultQuotaPolicy().getType(),
-                                    ((RequestCountLimit) limit).getRequestCount(),
-                                    policy.getDefaultQuotaPolicy().getLimit().getUnitTime(),
-                                    policy.getDefaultQuotaPolicy().getLimit().getTimeUnit(),
-                                    ((APIPolicy) policy).getPipelines(), API_TIER_LEVEL);
-                        }
-                    } else if (type.equals(PolicyConstants.BANDWIDTH_TYPE)) {
-                        if (limit instanceof BandwidthLimit) {
-                            addAPIPolicy(connection, policy.getPolicyName(), policy.getDisplayName(),
-                                    policy.getDescription(), policy.getDefaultQuotaPolicy().getType(),
-                                    ((BandwidthLimit) limit).getDataAmount(), limit.getUnitTime(), limit.getTimeUnit(),
-                                    ((APIPolicy) policy).getPipelines(), API_TIER_LEVEL);
-                        }
-                    }
-                } else {
-                    if (APIMgtConstants.ThrottlePolicyConstants.APPLICATION_LEVEL.equals(policyLevel)) {
-                        addApplicationPolicy(connection, policy.getPolicyName(), policy.getDisplayName(),
-                                policy.getDescription(), policy.getDefaultQuotaPolicy().getType(), 0, "",
-                                (int) policy.getDefaultQuotaPolicy().getLimit().getUnitTime(),
-                                policy.getDefaultQuotaPolicy().getLimit().getTimeUnit());
-                    } else if (APIMgtConstants.ThrottlePolicyConstants.SUBSCRIPTION_LEVEL.equals(policyLevel)) {
-                        addSubscriptionPolicy(connection, policy.getPolicyName(), policy.getDisplayName(),
-                                policy.getDescription(), policy.getDefaultQuotaPolicy().getType(), 0, "",
-                                (int) policy.getDefaultQuotaPolicy().getLimit().getUnitTime(),
-                                policy.getDefaultQuotaPolicy().getLimit().getTimeUnit());
-                    }
-                }
-            } catch (SQLException e) {
-                connection.rollback();
-                log.error(e.getMessage(), e);
-                throw new APIMgtDAOException(e);
+        String uuid = policy.getUuid();
+        if (StringUtils.isEmpty(uuid)) {
+            throw new APIMgtDAOException("Invalid argumants supplied. UUID is mandatory");
+        }
+
+        Connection connection;
+        try {
+            connection = DAOUtil.getConnection();
+            if (policy instanceof APIPolicy) {
+                updateAPIPolicy(uuid, connection, policy.getPolicyName(), policy.getDisplayName(),
+                        policy.getDescription(), policy.getDefaultQuotaPolicy().getType(), 0,
```

Comment:
Is this 0 is used set bandwidth based throttling. If yes add a TODO to update it with correct value